add/opensearch: connect with Python

laysauchoa opened the pull request:
Connect to OpenSearch cluster with Python

Signed-off-by: laysa.uchoa <laysa.uchoa@aiven.io>

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,47 @@
+Connect to OpenSearch cluster with Python
+=========================================
+
+One nice way to interact with your cluster is by using the `Python OpenSearch client <https://github.com/opensearch-project/opensearch-py>`_. 
```
I wonder if it was me who used word "nice" initially in the tutorial :D But let's remove it and make a neutral  statement. For example "You can interact with your OpenSearch cluster with the help of ...."

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,47 @@
+One nice way to interact with your cluster is by using the `Python OpenSearch client <https://github.com/opensearch-project/opensearch-py>`_. 
+It provides an easy syntax to send commands to your OpenSearch cluster. Follow its ``README`` file for installation instructions.
+
+In order to connect with the cluster, it is necessary to have the **Service URI** setting listed on the **Overview** page or can be accessed via the Aiven command line interface `service command <https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get>`_. ``service_uri`` contains credentials, therefore should be treated with care.
```
The sentence "In order to connect with the cluster, it is necessary to have the **Service URI** setting listed on the **Overview** page or can be accessed via the Aiven command line interface `service command <https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get>`_." seems to be a bit difficult to read. maybe something like this is better "To connect to you cluster, you will need the service URI of your OpenSearch cluster. You can find it in the  **Overview** page of the console [add link to our console main page]. Alternatively, you can access it via ...."

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,47 @@
+    
+    Check `documentation <https://github.com/opensearch-project/opensearch-py>`_ for more details.
+
+Congrats 🥳, you have a Python OpenSearch client class instance ready to be used to perform request operations and send responses in an easy way. 
```
I like all the smilies :D But I think according to our style guide - it should be a bit more neutral :)

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,47 @@
+
+Congrats 🥳, you have a Python OpenSearch client class instance ready to be used to perform request operations and send responses in an easy way. 
+
+Enjoy it!
```
Let's keep it out :) just to be aligned with the style of the rest of the docs :)

anelook reviewed (changes requested):
Nice and useful article :) Added couple of comments to align it with our writing styles.

laysauchoa commented:
thanks for suggestions @anelook addressed :)

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,45 @@
+=========================================
+
+You can interact with your cluster with the help of the `Python OpenSearch client <https://github.com/opensearch-project/opensearch-py>`_. 
+It provides an easy syntax to send commands to your OpenSearch cluster. Follow its ``README`` file for installation instructions.
```
I'd not use the word "easy", since what is easy for us, can be difficult for others. 
either remove it, or replace with something more neutral, like "convenient"

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,45 @@
+You can interact with your cluster with the help of the `Python OpenSearch client <https://github.com/opensearch-project/opensearch-py>`_. 
+It provides an easy syntax to send commands to your OpenSearch cluster. Follow its ``README`` file for installation instructions.
+
+To connect with your cluster, you need the **Service URI** of your OpenSearch cluster. You can find your cluster URI in the `**Overview** <https://console.aiven.io/project/dev-sandbox/services>`_ page. Alternatively, you can find it in Aiven command line interface `service command <https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get>`_. ``service_uri`` contains credentials, the ...
```
since we're repeating the same words twice, maybe replace some of it with "connection details"?

"To connect with your cluster, you need the **Service URI** of your OpenSearch cluster. You can find the connection details in ..."

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,45 @@
+You can interact with your cluster with the help of the `Python OpenSearch client <https://github.com/opensearch-project/opensearch-py>`_. 
+It provides an easy syntax to send commands to your OpenSearch cluster. Follow its ``README`` file for installation instructions.
+
+To connect with your cluster, you need the **Service URI** of your OpenSearch cluster. You can find your cluster URI in the `**Overview** <https://console.aiven.io/project/dev-sandbox/services>`_ page. Alternatively, you can find it in Aiven command line interface `service command <https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get>`_. ``service_uri`` contains credentials, the ...
```
sorry to be picky :D the phrase "you can find it in Aiven command line interface" seems a bit off, since you need to use the cli to get it. maybe something around " Alternatively, you can retrieve it by running `avn-service-get command<https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get>`_" in Aiven command line interface"

anelook replied:
also, since https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get is same website, better not to refer to https://developer.aiven.io, but to use an internal link. We use  ":doc:" and ":ref:" for relative paths.
P.S. I know this is probably copied from some of my stuff where I made same mistake in the past :D

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,45 @@
+
+    <https://<user>:<password>@<host>:<port>
+
+As you can see it contains sensitive information; therefore, 
```
This seem to be a long deviation about dotenv... Just a thought, but I'd try to limit it to bare mentioning. Since we don't want it to dominate this small article..

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,45 @@
+    
+    Check `documentation <https://github.com/opensearch-project/opensearch-py>`_ for more details.
+
+This is all you need and you have a Python OpenSearch client class instance ready to be used to perform request operations and send responses to your cluster.
```
I'd remove "This is all you need" , because it is a bit opinionated and this might be not true for a set up on someone else computer :D

anelook reviewed (commented):
Left several comments which can make the content even better :)

laysauchoa commented:
Thanks for the comments @anelook. I have addressed the comments here: https://github.com/aiven/devportal/pull/479/commits/16a93b09e36e0c37e1953af2db4987c6d471f1a4

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,44 @@
+You can interact with your cluster with the help of the `Python OpenSearch client <https://github.com/opensearch-project/opensearch-py>`_. 
+It provides a convenient syntax to send commands to your OpenSearch cluster. Follow its ``README`` file for installation instructions.
+
+To connect with your cluster, you need the **Service URI** of your OpenSearch cluster. You can find the connection details in the `**Overview** <https://console.aiven.io/project/dev-sandbox/services>`_ page. Alternatively, you can find it in Aiven command line interface `service command <https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get>`_. ``service_uri`` contains credential ...
```
There is a layout issue with **. also, the link is not leading to the overview page, but to the list of services...

laysauchoa replied:
thanks, replaced the link. can you clarify what is the layout issue?

anelook commented on docs/products/opensearch/howto/connect-with-python.rst:
```diff
@@ -0,0 +1,44 @@
+You can interact with your cluster with the help of the `Python OpenSearch client <https://github.com/opensearch-project/opensearch-py>`_. 
+It provides a convenient syntax to send commands to your OpenSearch cluster. Follow its ``README`` file for installation instructions.
+
+To connect with your cluster, you need the **Service URI** of your OpenSearch cluster. You can find the connection details in the `**Overview** <https://console.aiven.io/project/dev-sandbox/services>`_ page. Alternatively, you can find it in Aiven command line interface `service command <https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get>`_. ``service_uri`` contains credential ...
```
One more minor things - I think for readers who don't know our cli well, would be helpful to see which command they need. Now the link to cli brings to correct place, but the description is still wage. What is we use
```suggestion
To connect with your cluster, you need the **Service URI** of your OpenSearch cluster. You can find the connection details in the `**Overview** <https://console.aiven.io/project/dev-sandbox/services>`_ page. Alternatively, you can retrieve it by running command `service get <https://developer.aiven.io/docs/tools/cli/service.html#avn-service-get>`_ from Aiven command line interface.  ``service_uri`` contains credentials, therefore should be treated with care.
```

laysauchoa replied:
I brought the changes and changed to use ref.

anelook reviewed (changes requested):
all looks good, apart from a small layout issue! also while I'm at it, I added couple of other suggestions which makes easier for the reader.

anelook reviewed (approved):
Looks great! Thanks @laysauchoa :)
